### PR TITLE
How to collect update and organize python logs 

### PR DESCRIPTION
**Pull request by kymomolly**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5146 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by kymomolly** [comment]
@paulodhiambo , you can have the pr for reviewing

**Comment by kymomolly** [comment]
Hey @paulodhiambo , how is the progress

**Comment by kymomolly** [comment]
Helle @paulodhiambo , how is the progress of this pr

**Comment by kymomolly** [comment]
Hello @paulodhiambo , I'm inquiring about the tech tutorial label.

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+- **level**: plunging succession of events. The available log ranks are `DEBUG`, `INFO`, `WARNING`, `ERROR`, and `CRITICAL`. The default level is `WARNING`.
+- **controller**: determines where the application will pipe your logs.
+- **design**: messages are logged in the format: `LEVEL>:LOGGER NAME>: MESSAGE>`.
+  
+You could be losing out on truly low logs that can help with underlying cause evaluation, based on how the logging module provides `WARNING` and, more important, level logs as a replacement. 
+
+So instead, use a 'FileHandler' to log single or additional documents on the circle, maybe using a `StreamHandler` or a `SocketHandler` to route logs to the control center or an unessential help over the organization.
+
+#### Benefits of logging documents
+While streaming logs to an outside objective, your application doesn't need to reckon with the possibility of organization-related irregularities.
```
Comment (LinusMuema) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md:48]: For instance, the statement `while streaming logs to an outside objective...`. I had to pause my reading to first understand what `outside objective` stands for. You can make use of easy language and say `when showing logs in your application...`. Or however you find easy. 

A good exercise would be to ask "how easy would a 5 year old understand what I am saying?"

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+#### Benefits of logging documents
+While streaming logs to an outside objective, your application doesn't need to reckon with the possibility of organization-related irregularities. 
+
+Furthermore, in the event of any issues happening while at the same time streaming logs over the organization, you'll not lose admittance to those logs since we'll have upheld them on every server.
+
+It considers making a more customized logging setup, whereby you're ready to isolate files and concentrate those documents with a log following asset.
+  
+#### Representation of basicConfig
+Logs follow a specific design that includes the accompanying attributes:
```
Comment (LinusMuema) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md:55]: You can change `accompanying` to `following`...

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+- `%(asctime)s`: it yields the date and season of the log, in [local time](https://docs.python.org/3.7/library/time.html#time.asctime).
+- `%(rankname)s` : the logging rank of the information.
+- `%(purport)s`: tenor of the log. Read more about log attributes [here](https://docs.python.org/3/library/logging.html#logrecord-ascribes).
+
+### Make a redid arrangement with different loggers and objections.
+The more your application scales, the more you need to utilize a reliable and explained way of designing each logger by incorporating the log name as a log section. 
+
+Arrange various logs and record their personalities powerfully to self-assertively set the log title to suit the personality of your module. Utilize the logging library's incorporated [getLogger() approach](https://docs.python.org/3.7/library/logging.html#logging.getLogger):
```
Comment (LinusMuema) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md:63]: Do logs have `personalities`? Or did you mean `attributes / properties`?

**Review by LinusMuema — CHANGES_REQUESTED** [review]
Hi @kymomolly 

I have seen that this article has been through two closed PRs. Which means the article isn't doing well in the review process. I saw that @paulodhiambo had asked you to check your vocabulary which I have also noticed. I haven't gone through your entire article and I have had a hard time with various statements and paragraphs. 

Please go through already published content on section and take note of the language used. Simple and easy to go through.

**Comment by kymomolly** [comment]
Hello@LinusMuema , I close checked the grammar on the complex words and I have replaced them with clear and simpler words .

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md (hunk 3 above)
Comment (kymomolly) [reply content/articles/how-to-collect-update-and-organize-python-logs/index.md:63]: I meant properties

**Comment by kymomolly** [comment]
Hello @LinusMuema , I Will resolve the vocabularies issues

**Comment by kymomolly** [comment]
Hello @LinusMuema , I've done the requested changes. I hope its clear to understand.

**Comment by hectorkambow** [comment]
@kymomolly 
 
Good afternoon and thank you for submitting your topic suggestion. It seems that this PR is still quite a bit aways from being completed or publish ready. Some areas were a bit difficult to read/understand.
 
In an effort to be fair to our larger EngEd community as a rule of thumb - we can only accept content (pull requests) that are complete and fully polished articles/tutorial - meaning they are ready to be published. 
 
All revisions and edits **should be completed on your own forked repo** (as to not take up room in the queue) - this allows all revisions to be done on your forked personal repo. 
 
We will be closing this PR and please re-submit it when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. Be sure to go through the entire article again to proofread - and fix any issues that the reviewer may have missed (or has been already implemented). 
 
Our Peer Reviewer and Content Moderator teams do NOT provide any revisions services. All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.
 
>NOTE: A duplicate PR will go back into the queue and will not receive any further major revisions. A duplicate PR should be 100% ready to be published, otherwise it will be closed immediately. We encourage our new contributors to go over our [resources page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) and [new contributor folder](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) to be better familiar with the level of quality and style we are looking to publish.
 
Closing a PR does NOT mean the article can not be published (or fixed) in the future it simply means that the edits, revisions, and fixes will happen outside of the queue.


cc @paulodhiambo 

cc @LinusMuema

**Pull request closed without merging**